### PR TITLE
[Github-Actions] Windows CI

### PR DESCRIPTION
**Pull request by Anas35**
## Description

Setting up Github Actions for windows. RN, Only working for Battery_plus packages. 

Trying to reuse the same jobs for other packages integration test.

## Related Issues

#1094 

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: .github/workflows/battery_plus.yaml [hunk 1]
```diff
@@ -117,19 +117,40 @@ jobs:
...
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v2
+      - name: "Install Flutter"
+        shell: bash
+        run: ./.github/workflows/scripts/install-flutter.sh stable
+      - name: "Install Tools"
+        shell: bash
+        run: ./.github/workflows/scripts/install-tools.sh
+      - name: "Bootstrap Workspace"
```
Comment (Anas35) [inline .github/workflows/battery_plus.yaml:147]: We are getting dependencies all the packages for each integration test. Shouldn't we get dependencies  only for respective packages in integration tests?

File: .github/workflows/battery_plus.yaml [hunk 2]
```diff
@@ -117,19 +117,40 @@ jobs:
...
+        shell: bash
+        run: flutter pub global run melos bootstrap
+      - name: "Web Integration Test"
```
Comment (Anas35) [inline .github/workflows/battery_plus.yaml:150]: nit: windows

File: .github/workflows/battery_plus.yaml [hunk 3]
```diff
@@ -117,19 +117,40 @@ jobs:
...
+        uses: actions/checkout@v2
+      - name: "Install Flutter"
+        shell: bash
+        run: ./.github/workflows/scripts/install-flutter.sh stable
+      - name: "Install Tools"
+        shell: bash
+        run: ./.github/workflows/scripts/install-tools.sh
+      - name: "Bootstrap Workspace"
+        shell: bash
+        run: flutter pub global run melos bootstrap
```
Comment (Anas35) [inline .github/workflows/battery_plus.yaml:149]: [see](https://github.com/invertase/melos/issues/241). maybe, In windows it can't set global env var.

**Comment by miquelbeltran** [comment]
I set it to draft while it is WIP, feel free to remove the draft status when ready to review

**Comment by Anas35** [comment]
Look like few integration test are failing in windows. Looking into it.

**Comment by miquelbeltran** [comment]
Could you run the CI jobs on your own fork and then open a PR when ready? that way we are not using all VM instances in this project

**Comment by Anas35** [comment]
Extremely sorry, Look like I caused quite a lot of trouble. Once again, Sorry for the inconvenience. 

Look like someone has opened an PR for this same issue, I don't know what is the process for different PR for one issue. Will close this one and keep a eye on that PR.

**Comment by miquelbeltran** [comment]
We ended up accepting the other PR for the Windows CI work, as it was more complete and provided multiple fixes. There's still other opportunities to contribute during Hacktoberfest!

**Pull request closed without merging**